contracts/governance/extensions/GovernorOverrideDelegateVote.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {GovernorVotes} from "./GovernorVotes.sol";
+import {IVotes} from "../utils/IVotes.sol";
+import {IERC5805} from "../../interfaces/IERC5805.sol";
+import {SafeCast} from "../../utils/math/SafeCast.sol";
+import {Time} from "../../utils/types/Time.sol";
+import {VotesOverridable} from "../utils/VotesOverridable.sol";
+
+/**
+ * @dev Extension of {Governor} which enables delegatees to override the vote of their delegates. This module requires a
+ * token token that inherits `VotesOverridable`.
+ */
+abstract contract GovernorOverrideDelegateVote is GovernorVotes {
+    /**
+     * @dev Supported vote types. Matches Governor Bravo ordering.
+     */
+    enum VoteType {
+        Against,
+        For,
+        Abstain
+    }
+
+    struct VoteReceipt {
+        bool hasVoted;
+        uint8 support;
+    }
+
+    struct ProposalVote {
+        uint256 againstVotes;
+        uint256 forVotes;
+        uint256 abstainVotes;
+        mapping(address voter => VoteReceipt) voteReceipt;
+        mapping(address voter => VoteReceipt) overrideVoteReceipt;
```

Comment:
Nobody is going to override the override ... so we should not need to track the support of the override, do we?
Feels to me that the override part of the Receipt only needs to be a boolean.

Also, could we use a single mapping for receipt, with the receipt structure being
- bool hasVoted
- uint8 voteSupport
- bool hasOverriden
- uint256 (or smaller?) overrideWeight

That would give better datalocality when we move to verkle.

Note, you could even merge hasVoted + support by storing `support + 1` (here we control that support is never going to be 255, so overflow are not an issue)

Reply:
> Nobody is going to override the override ... so we should not need to track the support of the override, do we? Feels to me that the override part of the Receipt only needs to be a boolean.

Yes this is correct. Also, given that snapshots within `Votes` are stored as `uint208`, we only need to allocate that to `overrideWeight`. So we could fit it all into one slot.